src/securemul.rs

```diff
@@ -0,0 +1,430 @@
+use crate::error::BoxError;
+use crate::field::Field;
+use crate::helpers::ring::{HelperAddr, Ring};
+use crate::prss::Participant;
+use crate::replicated_secret_sharing::ReplicatedSecretSharing;
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+use thiserror::Error;
+
+/// Secure multiplication protocol using replicated secret sharing over some field `F`.
+/// * [paper](https://eprint.iacr.org/2018/387.pdf)
+#[derive(Debug)]
+pub struct SecureMul<F> {
+    index: u128,
+    a_share: ReplicatedSecretSharing<F>,
+    b_share: ReplicatedSecretSharing<F>,
+}
+
+/// A message sent by each helper when they've multiplied their own shares
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct DValue {
+    index: u128,
+    d: u128,
```

Comment:
Woah... I don't think we need anything that big. Honestly - even a u32 is probably going to be enough for our usecase.

Reply:
It should be determined by `F`

Reply:
Fair enough.